cranelift/codegen/src/machinst/compile.rs

```diff
@@ -14,7 +14,7 @@ use regalloc2::{self, MachineEnv};
 pub fn compile<B: LowerBackend + TargetIsa>(
     f: &Function,
     b: &B,
-    abi: Box<dyn ABICallee<I = B::MInst>>,
+    abi: AbiCallee<<<B as LowerBackend>::MInst as MachInst>::ABIMachineSpec>,
```

Comment:
The type checker was complaining it couldn't infer where the `MInst` projection was coming from, not sure why it can't be a little smarter here, but alas.